tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -1009,7 +1011,8 @@ def ensure_runner_stopped(self):
             self.logger.debug("Runner process exited with code %i" % self.test_runner_proc.exitcode)
 
     def runner_teardown(self):
```

Comment:
I not sure in what scenario we will need `runner_teardown`. Looks like this is triggered either when there is an exception thrown during setup, or when the TestRunner received a 'stop' command [1]. In both case `child_stop_flag` will be set, and `TestRunner. teardown` will be called.

OK, looks like no exception will be thrown during `TestRunner.run` [2]. We should remove the misleading code.

Then since it is always `TestRunnerManager` to tell `TestRunner` to stop. The message is not necessary.

[1] https://source.chromium.org/chromium/chromium/src/+/main:third_party/wpt_tools/wpt/tools/wptrunner/wptrunner/testrunner.py;l=140
[2] https://source.chromium.org/chromium/chromium/src/+/main:third_party/wpt_tools/wpt/tools/wptrunner/wptrunner/testrunner.py;l=145

Reply:
> I not sure in what scenario we will need runner_teardown

Right, it's essentially a signal that the runner process is about to exit, but I've concluded it's safe to remove. There are two scenarios to consider:

1. A normal `stop` message sent from the main process to the runner (e.g., for a browser restart). The resulting `runner_teardown` message [is ignored](https://source.chromium.org/chromium/chromium/src/+/main:third_party/wpt_tools/wpt/tools/wptrunner/wptrunner/testrunner.py;l=1037-1041;drc=93525f9e916a879d2805b1a2b419906ec17dda5f;bpv=0;bpt=0). Note that the child stop flag is not set in this case, since that breaks the `TestRunnerManager` thread out of its loop, permanently reducing the number of workers by one.
1. An unexpected top-level exception in `start_runner()` (e.g., `KeyboardInterrupt`). That always sets the stop flag, which guarantees that the [`wait_event` loop and main run loop][1] terminate, so there's no need for `wait_event` to explicitly transition to the `RunnerManagerState.stop` state.

[1]: https://source.chromium.org/chromium/chromium/src/+/main:third_party/wpt_tools/wpt/tools/wptrunner/wptrunner/testrunner.py;l=468-469;drc=93525f9e916a879d2805b1a2b419906ec17dda5f;bpv=0;bpt=0

> OK, looks like no exception will be thrown during TestRunner.run [2]. We should remove the misleading code.

Done